optuna/version.py

```diff
@@ -1 +1 @@
-__version__ = "2.4.0"
+__version__ = "2.5.0.dev"
```

Comment:
What are the rules for including a _dot_ here? I noticed some frameworks do, but I also remember discussing whether to include a _hyphen_ before with @toshihikoyanase when releasing the [RC for v2.0](https://pypi.org/project/optuna/2.0.0rc0). We decided then based on other frameworks including TensorFlow, PyTorch, and other sources (if I remember correctly) to 

1. Exclude the hyphen from the versioning https://github.com/optuna/optuna/blob/v2.0.0-rc0/optuna/version.py
2. Include the hyphen in the branch name/tag https://github.com/optuna/optuna/releases/tag/v2.0.0-rc0

Should we apply this rule here as well?

Reply:
It's seemingly covered by PEP and should be included https://www.python.org/dev/peps/pep-0440/#public-version-identifiers

Reply:
Thanks for the careful consideration and investigation. I followed `pytorch-ignite` which releases very frequently such as `0.2.0.dev2020xxxx`. So, the current PR's versioning is fine according to the PeP 440?

Reply:
Yup, it's complying.

From the PEP.

> Development release separators
Development releases allow a ., -, or a _ separator as well as omitting the separator all together. The normal form of this is with the . separator. This allows versions such as 1.2-dev2 or 1.2dev2 which normalize to 1.2.dev2.

> Implicit development release number
Development releases allow omitting the numeral in which case it is implicitly assumed to be 0. The normal form for this is to include the 0 explicitly. This allows versions such as 1.2.dev which is normalized to 1.2.dev0.